src/managers/IntegrationManager.ts

```diff
@@ -0,0 +1,60 @@
+import SoftwareIntegration from "../model/SoftwareIntegration";
+import { getIntegrations, syncIntegrations } from "./FileManager";
+
+const { WebClient } = require("@slack/web-api");
+
+export async function updateSlackIntegrations(user) {
+  if (user && user.integrations) {
+    return await updateIntegrations(user, "slack");
+  }
+  return false;
+}
+
+export async function updateSpotifyIntegrations(user) {
+  if (user && user.integrations) {
+    return await updateIntegrations(user, "spotify");
+  }
+  return false;
+}
+
+async function updateIntegrations(user, name) {
+  let foundNewIntegration: boolean = false;
+  let currentIntegrations = getIntegrations();
+  for (const integration of user.integrations) {
+    // {access_token, name, plugin_uuid, scopes, pluginId, authId, refresh_token, scopes}
+    if (integration.name.toLowerCase() === name && integration.status.toLowerCase() === "active") {
```

Comment:
could you simplify this nesting a little bit if you did something like:
```javascript
const isActive = integration.name.toLowerCase() === name && integration.status.toLowerCase() === "active";
const isFound = currentIntegrations.find((n) => n.authId === integration.authId;
if (isActive && !isFound) {
   // do all the things
}
```

Reply:
yeah, totally. easier to read. i'll update it to that

Reply:
updated